docs/secret-rotation.md

```diff
@@ -0,0 +1,254 @@
+#### [Deleting Old Secrets](#deleting-old-secrets-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - Some housekeeping steps post secret rotation.
+#### [Rotating Compromised Secrets](#rotating-compromised-secrets-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - Summary for the rotation of secrets posing a security threat.
+
+## [**Validate Configuration**](#validate-configuration-link)
+#### [Sonobuoy](#sonobuoy-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - Brief outline on validating the Kubernetes configuration.
+
+#
+<a id="pre-requisites-link"></a>
+# **Pre-Requisites**
+
+This document assumes that a self-managed CAPI cluster deployed with FluxCD is already configured and deployed, if not, one can be deployed by following the instructions found in [this](https://github.com/stackhpc/capi-helm-fluxcd-config) GitHub repository.
+
+In addition to this, the user trying to rotate the secrets should have an account on Horizon with the necessary permissions.
+
+<a id="useful-resources-link"></a>
+### **Useful Resources**
+
+[CAPI cluster FluxCD repository](https://github.com/stackhpc/capi-helm-fluxcd-config)
+
+[Waldur FluxCD Demo repository](https://github.com/stackhpc/waldur-fluxcd-gitops-demo)
+
+[CAPI Cluster debugging manual](https://github.com/azimuth-cloud/capi-helm-charts/blob/main/charts/openstack-cluster/DEBUGGING.md)
+
+<a id="applications--software-link"></a>
+### **Applications & Software**
+
+| This document assumes that the Kubernetes cluster’s ```kubeconfig``` is known. |
+| :---- |
+
+Kubeseal
+
+Current CAPI cluster FluxCD config
+
+Horizon account
+
+k9s (for debugging, optional)
+
+<a id="outline-link"></a>
+# **Outline**
+
+The overall idea is to take advantage of Kubeseal’s encryption, allowing us to upload secrets to our repositories; meaning that after a branch containing the new secrets is merged into the branch being tracked by FluxCD, the FluxCD GitOps flow will update the current cluster deployment to reflect the newly merged branch.
+
+Although the basic outline will remain the same, there are some variations in the preliminary actions depending on the situation and reason why secrets are being rotated.
+
+The two main categories fall under **cluster maintenance** and **security threat**, and the differences in these situations will be further explained below.
+
+<a id="cluster-maintenance-link"></a>
+### **Cluster Maintenance**
+
+It is often a good idea and good security practice to rotate passwords and secrets, as it reduces the chances of them existing long enough for them to be exploited or end up in the wrong hands; often many secrets will have an expiration date precisely for this reason.
+
+Therefore, in the case of the user simply updating secrets for the purpose of cluster maintenance it is simply enough to replace the current Kubeseal encrypted secret with an updated, manually encrypted, Kubeseal secret via a GitHub pull request and merge. Then delete the old application credential from Horizon.
```

Comment:
Nit: Personally I'd format all references to the `kubeseal` tool as lower case inside backticks, as they do in the official [docs](https://github.com/bitnami-labs/sealed-secrets), but maybe that's just me...
```suggestion
Therefore, in the case of the user simply updating secrets for the purpose of cluster maintenance it is simply enough to replace the current Kubeseal encrypted secret with an updated, manually encrypted, `kubeseal` secret via a GitHub pull request and merge. Then delete the old application credential from Horizon.
```